h5netcdf/tests/test_h5netcdf.py

```diff
@@ -11,10 +14,35 @@
 
 from pytest import raises
 
+try:
+    import h5pyd
+    without_h5pyd = False
+except ImportError:
+    without_h5pyd = True
+
+
+@pytest.fixture(params=['testfile.nc', 'hdf5://testfile'])
+def tmp_netcdf(request, tmpdir):
```

Comment:
Rather than adding lots of `pytest.skip` invocations below (which is a little ugly), I would suggest writing two different fixtures:
- `tmp_local_netcdf`: like the original version of `tmp_netcdf`
- `tmp_local_or_remote_netcdf`: parameterized to either be a local or remote URL (i.e., this version of `tmp_netcdf`)

Then each test can simply use the appropriate fixture.